examples/hello_compute_c/main.c

```diff
@@ -0,0 +1,139 @@
+
+    WGPUAdapterId adapter = wgpu_instance_get_adapter(instance,
+        &(WGPUAdapterDescriptor){
+            .power_preference = WGPUPowerPreference_LowPower,
+        });
+
+    WGPUDeviceId device = wgpu_adapter_request_device(adapter,
+        &(WGPUDeviceDescriptor){
+            .extensions =
+                {
+                    .anisotropic_filtering = false,
+                },
+        });
+
+	uint8_t *staging_memory;
+
+    WGPUBufferId staging_buffer = wgpu_device_create_buffer_mapped(device,
+        &(WGPUBufferDescriptor){.size = numbers_length,
+            .usage = WGPUBufferUsage_MAP_READ | WGPUBufferUsage_TRANSFER_DST |
+                WGPUBufferUsage_TRANSFER_SRC},
+            &staging_memory);
+
+    WGPUBufferId storage_buffer = wgpu_device_create_buffer(device,
+        &(WGPUBufferDescriptor){.size = size,
+            .usage = WGPUBufferUsage_STORAGE | WGPUBufferUsage_TRANSFER_DST |
+                WGPUBufferUsage_TRANSFER_SRC});
+
+    WGPUBindGroupLayoutId bind_group_layout =
+        wgpu_device_create_bind_group_layout(device,
+            &(WGPUBindGroupLayoutDescriptor){
+                .bindings = &(WGPUBindGroupLayoutBinding){.binding = 0,
+                    .visibility = WGPUShaderStage_COMPUTE,
+                    .ty = WGPUBindingType_StorageBuffer}});
+
+	WGPUBindingResource resource = (WGPUBindingResource) {
+        .tag = WGPUBindingResource_Buffer,
+        .buffer = (WGPUBindingResource_WGPUBuffer_Body) {
+            storage_buffer
+        }};
+
+    WGPUBindGroupId bind_group = wgpu_device_create_bind_group(device,
+            &(WGPUBindGroupDescriptor){.layout = bind_group_layout,
+                .bindings = &(WGPUBindGroupBinding){
+                    .binding = 0, .resource = resource}});
+
+    WGPUPipelineLayoutId pipeline_layout =
+        wgpu_device_create_pipeline_layout(device,
+            &(WGPUPipelineLayoutDescriptor){
+                .bind_group_layouts = &bind_group_layout});
+
+    WGPUShaderModuleId shader_module = wgpu_device_create_shader_module(device,
+        &(WGPUShaderModuleDescriptor){
+            .code = read_file("./../../data/shader.comp.spv")});
+
+    WGPUComputePipelineId compute_pipeline =
+        wgpu_device_create_compute_pipeline(device,
+            &(WGPUComputePipelineDescriptor){.layout = pipeline_layout,
+                .compute_stage = (WGPUPipelineStageDescriptor){
+                    .module = shader_module, 
+					.entry_point = "main"}});
+
+    WGPUCommandEncoderId encoder = wgpu_device_create_command_encoder(
+        device, &(WGPUCommandEncoderDescriptor){.todo = 0});
+
+    wgpu_command_buffer_copy_buffer_to_buffer(
+        encoder, staging_buffer, 0, storage_buffer, 0, size);
+
+    WGPUComputePassId command_pass =
+        wgpu_command_encoder_begin_compute_pass(encoder);
+    wgpu_compute_pass_set_pipeline(command_pass, compute_pipeline);
+
+    wgpu_compute_pass_set_bind_group(command_pass, 0, bind_group, 0, 0);
```

Comment:
nit: this is minor but we should use `NULL` instead of `0` for the `offsets` argument, like `wgpu_compute_pass_set_bind_group(command_pass, 0, bind_group, NULL, 0);`